### PR TITLE
Minor refactoring attribute checkers

### PR DESCRIPTION
**Pull request by shendiaomo**
1. Add lower/upper bound checkers for convenience
1. Remove all the newInt or newFloat utility functions
1. Modify boolean arithmetic to make code slightly clearer

### CODE REVIEW

File: pkg/sql/codegen/attribute/checker.go [hunk 1]
```diff
@@ -17,87 +17,90 @@ import (
...
+// IntRangeChecker is a helper function to generate range checkers on attributes.
 // lower/upper indicates the lower bound and upper bound of the attribute value.
-// If lower/upper is nil, it means no boundary.
 // includeLower/includeUpper indicates the inclusion of the bound.
-func IntRangeChecker(lower, upper *int, includeLower, includeUpper bool) func(interface{}) error {
-	checker := func(e interface{}) error {
-		i, ok := e.(int)
-		if !ok {
-			return fmt.Errorf("expected type float32, received %T", e)
+func IntRangeChecker(lower, upper int, includeLower, includeUpper bool) func(interface{}) error {
```
Comment (Yancey0623) [inline pkg/sql/codegen/attribute/checker.go:67]: I think lower and upper should be a pointer, we can use `nil` to represent no-bounds: `(1, nil, true, false)` => `[1, ~)`

Comment (shendiaomo) [reply pkg/sql/codegen/attribute/checker.go:67]: This change uses LowerBoundChecker and UpperboundChecker to represent infinity bounds. The two functions behave similarly to the builtin operators `>=`, `<` etc. As a result, we don't need `nil` anymore.

File: pkg/sql/codegen/attribute/checker.go [hunk 2]
```diff
@@ -17,87 +17,90 @@ import (
...
-	return checker
 }
 
-func newInt(i int) *int {
-	return &i
+// Float32UpperBoundChecker returns a range checker that only checks the upper bound.
+func Float32UpperBoundChecker(upper float32, includeUpper bool) func(interface{}) error {
+	return func(attr interface{}) error {
+		if f, ok := attr.(float32); ok {
+			if f < upper || includeUpper && f == upper {
```
Comment (wangkuiyi) [inline pkg/sql/codegen/attribute/checker.go]: As explained in https://bitbashing.io/comparing-floats.html, it doesn't make much sense to compare the equality of two float values. We should rewrite the conditional something like

```go
if (!includeUpper && f < upper) || (includeUpper && f <= upper) {
    return nil
}
```

**Review by wangkuiyi — DISMISSED** [review]

File: pkg/sql/codegen/attribute/checker.go (hunk 2 above)
Comment (shendiaomo) [reply pkg/sql/codegen/attribute/checker.go]: Done.

**Review by wangkuiyi — APPROVED** [review]

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**